aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <cstdint>
+#include <utility>
+
+#include "cuda_runtime_api.h"
+
+#include <ATen/ATenGeneral.h>
+#include <ATen/Error.h>
+
+/*
+* A CUDA event interface with no CUDA build dependency.
+*
+* Includes the CUDAEvent RAII class and a pointer-based event API.
+*/
+
+struct CUDAEventInternals;
+
+namespace at {
+namespace cuda {
+
+struct CUDAStream;
+
+namespace detail {
+
+// Pointer-based API (for internal use)
+// Note: ATen/Context is preferred to work with streams safely
+AT_API CUDAEventInternals* CUDAEvent_create(unsigned int flags);
```

Comment:
Is there a reason not to use the pImpl pattern? It is more a more idiomatic for hiding the implementation of a class

Reply:
I just followed whatever we use for CUDA streams. This is pretty much isomorphic to pImpl.